packages/experimental/Text/src/Text.tsx

```diff
@@ -39,56 +42,45 @@ export type TextProps<TBase = ITextProps> = TBase &
     disabled?: boolean;
   };
 
-/**
- * These are the tokens which are also present in props. If specified in props this will override the values
- * from tokens.
- */
-const tokensThatAreAlsoProps: (keyof TextTokens)[] = ['variant', 'color'];
+const useTextTokens = buildUseTokens<TextTokens>(
+  (t) => ({
+    variant: 'secondaryStandard',
+    color: t.colors.bodyText,
+    disabled: {
+      color: t.colors.disabledText,
+    },
+  }),
+  textName,
+);
 
-/** Type to use for compose */
-interface TextType {
-  props: TextProps;
-  slotProps: { root: ITextProps };
-  tokens: TextTokens;
-}
+export const Text = compressible<TextProps, TextTokens>((props: TextProps, useTokens: UseTokens<TextTokens>) => {
+  // split out color and variant from props
+  const { color, variant, style, disabled, ...rest } = props;
+  const theme = useFluentTheme();
+  // get the tokens from the theme
+  let [tokens, cache] = useTokens(theme);
 
-export const Text = compose<TextType>({
-  displayName: textName,
-  /** Settings for the use-styling hook */
-  tokens: [
-    (t) => ({
-      variant: 'secondaryStandard',
-      color: t.colors.bodyText,
-      disabled: {
-        color: t.colors.disabledText,
-      },
+  // apply states like disabled if specified in props
+  [tokens, cache] = applyTokenLayers(tokens, ['disabled'], cache, (state) => props[state]);
+  // override variant or color from props
+  [tokens, cache] = patchTokens(tokens, cache, { color, variant });
+
+  // now build the text style from tokens that can be shared between different Text instances
+  const [tokenStyle] = cache(
+    () => ({
+      margin: 0,
+      color: tokens.color,
+      ...fontStyles.from(tokens, theme),
     }),
-    textName,
-  ],
-  states: ['disabled'],
-  tokensThatAreAlsoProps,
-  slotProps: {
-    root: buildProps<ITextProps, TextTokens>(
-      (tokens: TextTokens, theme: Theme) => ({
-        style: {
-          margin: 0,
-          color: tokens.color,
-          ...fontStyles.from(tokens, theme),
-        },
-      }),
-      ['color', ...fontStyles.keys],
-    ),
-  },
-  /** Settings for the useSlots that will be passed on */
-  slots: { root: RNText },
-  filters: { root: filterTextProps },
-  /** render function for the component */
-  render: (props: TextProps, useSlots: UseSlots<TextType>) => {
-    // stage one, execute any hooks, styling lookups to build the styled slot
-    const Root = useSlots(props).root;
-    // return a function used to complete the render
-    return (rest: TextProps, children: React.ReactNode) => <Root {...mergeProps(props, rest)}>{children}</Root>;
-  },
-});
+    [],
+  );
```

Comment:
Is the type of thing that buildUseStyling / slotProps were doing behind the scenes for us within `compose` that we now must do manually with `compressible`?

Reply:
Yep.

- `states: ['disabled']` - this causes the `applyTokenLayers` call to happen
- `tokensThatAreAlsoProps` - causes the equivalent of `patchTokens`
- `slotProps: { root: buildProps< ...` - is the equivalent to the `const [tokenStyle] = cache(` call.
- `filters: { root: filterTextProps }` - is really only required because we aren't splitting the non-text props out of the input props before forwarding them on.

For many components the `applyTokenLayers` or `patchTokens` calls aren't required. In terms of caching styles that is an important concept to understand but the helpers were just as much code as the direct code so they end up being just extra things to learn.

Reply:
The rewrite is functionally the same and produces the same render tree, with the exception of not supporting `compose` and the things that can be done via compose.

Reply:
Thanks! I like that this example has comments that broke into steps the levels of layering that are happening. I think that will definitely be helpful in the future for also understanding how `compose` layers tokens and styles onto a component.